pkg/eventmonitor/consumers/process.go

```diff
@@ -0,0 +1,171 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux || windows
+
+package consumers
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/DataDog/datadog-agent/pkg/eventmonitor"
+	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
+)
+
+// ProcessCallback is a callback function that will be called when a process exec/exit event is received
+// Defined as an alias to avoid type incompatibilities
+type ProcessCallback = func(uint32)
+
+// ProcessConsumer represents a consumer of process exec/exit events that can be subscribed to
+// via callbacks
+type ProcessConsumer struct {
+	// id is the identification of this consumer in the event data stream
+	id string
+
+	// chanSize is the channel size that will be used to send events to this consumer from the event data stream
+	chanSize int
+
+	// execCallbacks holds all subscriptors to process exec events
+	execCallbacks callbackMap
+
+	// exitCallbacks holds all subscriptors to process exit events
+	exitCallbacks callbackMap
+}
+
+// event represents the attributes of the generic eventmonitor type that we will copy and use in our process consumer
+type event struct {
```

Comment:
isn't this a duplicate of some other struct already defined in event_monitor (or security/event_monitor)?

Reply:
same as the previous comment, if this is a specific consumer, move it out from the eventmonitor pkg. if that is a generic consumer, part of the event_monitor platform, i think we should stick to the model pkg with defined event struct there

Reply:
I'm not sure I follow, I might be misunderstanding how the consumer should work. My understanding is that the CWS probe will call the [consumer.Copy](https://github.com/DataDog/datadog-agent/blob/guillermo.julian/eventmonitor-common-process-consumer/pkg/security/probe/probe.go#L315), which should return a copied object that will later be sent to the consumer channel. In that case it's more efficient to have a type where we copy only the fields we need rather than every field in the `model.Event/BaseEvent` types.